### PR TITLE
fix: [DEL-5259]: use nanoid to get short uuid

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
### Summary
**The problem we tried to solve is to produce a UUID that is shorter than 36 characters which is generally the length produced by the UUID library. So we used nano id which has the same probability of collision as uuid but gives less number of characters (~21), which solves our purpose here.**

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
     "monaco-editor-core": "0.15.5",
     "monaco-languages": "1.6.0",
     "monaco-plugin-helpers": "^1.0.2",
+    "nanoid": "^4.0.0",
```
Comment (abhinavrastogi-harness) [inline package.json:100]: please get approval from sec team to add this new library, and link the ticket here.

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]
need sec review for new lib

**Comment by udayvunnam** [comment]
Can we also have a follow-up ticket that can remove the UUID library and use nanoId at all places?

File: package.json (hunk 1 above)
Comment (swarajcheguri-harness) [reply package.json:100]: Here is the ticket.https://harness.atlassian.net/browse/SECOPS-4917

**Comment by swarajcheguri-harness** [comment]
> Can we also have a follow-up ticket that can remove the UUID library and use nanoId at all places?

https://harness.atlassian.net/browse/PL-29791

File: package.json (hunk 1 above)
Comment (swarajcheguri-harness) [reply package.json:100]: > need sec review for new lib
Approved by SECOP
https://harness.atlassian.net/browse/SECAPP-593

**Review by abhinavrastogi-harness — DISMISSED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**